homeassistant/components/ovo_energy/sensor.py

```diff
@@ -32,199 +52,148 @@ async def async_setup_entry(
 
     if coordinator.data:
         if coordinator.data.electricity:
-            entities.append(OVOEnergyLastElectricityReading(coordinator, client))
-            entities.append(
-                OVOEnergyLastElectricityCost(
+            entities = [
+                *entities,
+                OVOEnergySensor(
+                    coordinator,
+                    OVOEnergySensorEntityDescription(
+                        key=f"{DOMAIN}_{client.account_id}_last_electricity_reading",
+                        name="OVO Last Electricity Reading",
+                        device_class=DEVICE_CLASS_ENERGY,
+                        state_class=STATE_CLASS_TOTAL_INCREASING,
+                        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+                        value=lambda usage: usage.electricity[-1].consumption,
+                    ),
+                    client,
+                ),
+                OVOEnergySensor(
+                    coordinator,
+                    OVOEnergySensorEntityDescription(
+                        key=f"{DOMAIN}_{client.account_id}_last_electricity_cost",
+                        name="OVO Last Electricity Cost",
+                        device_class=DEVICE_CLASS_MONETARY,
+                        state_class=STATE_CLASS_TOTAL_INCREASING,
+                        native_unit_of_measurement=coordinator.data.electricity[
+                            -1
+                        ].cost.currency_unit,
```

Comment:
This one is a bit tricky. As the description is a dataclass, you can change the value later.
I would try something like this here (in `async_setup_entry`):
```py
for desc in (*SENSOR_TYPES_ELECTRICITY, * SENSOR_TYPES_GAS):
    if desc.key == "last_electricity_cost":
        desc.native_unit_of_measurement = coordinator.data.electricity[-1].cost.currency_unit
    elif desc.key == "last_gas_cost":
        ...
```